beacon-chain/types/block.go

```diff
@@ -31,18 +31,14 @@ type Block struct {
 // Return block with default fields if data is nil.
 func NewBlock(data *pb.BeaconBlock) *Block {
 	if data == nil {
+		//It is assumed when data==nil, you're asking for a Genesis Block
```

Comment:
What if someone wants NewBlock with input nil, but with current timestamp?

Reply:
2.1 spec doesn't call out timestamp, that's something we use internally for processing block and tracking block validity.

Reply:
Ops I messed up on the comments. What was your last comment?

Reply:
now I messed up the comments... my apologies.

I wanted to say that 2.1 spec doesn't require a timestamp anywhere except for the genesis block so i'm inclined to say that for those cases, people can manually alter the timestamp in the same manner i did it in the NewGenesisBlock function.

Reply:
That will work. My main concern is other services (ex: simulator/sync) uses NewBlock and relies on timestamp to advance & now they need to manually enter the time stamp or things might break. I'll wait for others to comment. Other than that, everything looks good. Thanks!

Reply:
If you can point me to places in the code where you think it would be problematic i'd be happy to take a look and maybe come up with some ideas/something cleaner... I've looked at simulator/service.go and it seems like they're creating their own timestamp there.

Reply:
I did more research on how `NewBlock` is used, and this looks good 👍 Sorry for my over concern 😅